handler/oauth2/flow_generic_code_token.go

```diff
@@ -0,0 +1,267 @@
+// Copyright © 2024 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package oauth2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ory/fosite/storage"
+
+	"github.com/ory/x/errorsx"
+
+	"github.com/ory/fosite"
+)
+
+// AccessRequestValidator handles various validations in the access request handling.
+type AccessRequestValidator interface {
+	// CanHandleRequest validates if the access request should be handled.
+	CanHandleRequest(requester fosite.AccessRequester) bool
+
+	// ValidateGrantTypes validates the grant types in the access request.
+	ValidateGrantTypes(requester fosite.AccessRequester) error
+
+	// ValidateRedirectURI validates the redirect uri in the access request.
+	ValidateRedirectURI(accessRequester fosite.AccessRequester, authorizeRequester fosite.Requester) error
+
+	// GetGrantType retrieves the grant type from the request.
+	GetGrantType(requester fosite.AccessRequester) fosite.GrantType
+}
+
+// CodeHandler handles authorization/device code related operations.
+type CodeHandler interface {
```

Comment:
As the device auth code grant has two codes (user & device), the naming should be clear at any point which one is used. Here I'm struggling really to understand which one it is.

Reply:
I strongly agree with this, I've been trying to follow along with an implementation (outside of Hydra), and it's been extremely unclear what the correct steps to handle user_code's are.